x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/optimizer/OptimizerTests.java

```diff
@@ -175,4 +178,51 @@ public void testWildcardEscapes() {
         assertEquals(like.pattern().asLuceneWildcard(), "* %bar_ * \\\\ \n \r \t");
         assertEquals(like.pattern().asIndexNameWildcard(), "* %bar_ * \\ \n \r \t");
     }
+
+    public void testCombineHeadBigHeadSmall() {
+        checkOffsetAndLimit(accept("process where true | head 10 | head 1"), 0, 1);
+    }
+
+    public void testCombineHeadSmallHeadBig() {
+        checkOffsetAndLimit(accept("process where true | head 1 | head 12"), 0, 1);
+    }
+
+    public void testCombineTailBigTailSmall() {
+        checkOffsetAndLimit(accept("process where true | tail 10 | tail 1"), 0, -1);
+    }
+
+    public void testCombineTailSmallTailBig() {
+        checkOffsetAndLimit(accept("process where true | tail 1 | tail 12"), 0, -1);
+    }
+
+    public void testCombineHeadBigTailSmall() {
+        checkOffsetAndLimit(accept("process where true | head 10 | tail 7"), 3, 7);
+    }
+
+    public void testCombineTailBigHeadSmall() {
+        checkOffsetAndLimit(accept("process where true | tail 10 | head 7"), 3, -7);
+    }
+
+    public void testCombineTailSmallHeadBig() {
+        checkOffsetAndLimit(accept("process where true | tail 7 | head 10"), 0, -7);
+    }
+
+    public void testCombineHeadBigTailBig() {
+        checkOffsetAndLimit(accept("process where true | head 1 | tail 7"), 0, 1);
+    }
+
+    public void testCombineHeadTailWithHeadAndTail() {
+        checkOffsetAndLimit(accept("process where true | head 10 | tail 7 | head 5 | tail 3"), 5, 3);
+    }
+
+    public void testCombineTailHeadWithTailAndHead() {
+        checkOffsetAndLimit(accept("process where true | tail 10 | head 7 | tail 5 | head 3"), 5, -3);
```

Comment:
Is the offset here `5` or `4`?

Reply:
command  | limit, offset
tail 10 -> -10  | 0
head 7  -> -7   | 3
tail 5  -> -5   | 3
head 3  -> -3   | 5